inc/endpoints/class-post-select-controller.php

```diff
@@ -140,7 +140,12 @@ public function get_items( $request ) {
 
 		$query_args = apply_filters( 'hm_gb_tools_post_select_query_args', $query_args );
 
+		do_action( 'hm_gb_tools_post_select_before_query', $query_args );
```

Comment:
I'm curious what benefit we get from an action that we don't get from the existing `hm_gb_tools_post_select_query_args` filter. Can you provide an example of a side-effect you'd want to trigger here, which couldn't be done by adding code to that filter we already have?

Reply:
Maybe it just feels like not using the filter for its correct purpose? I think it's probably OK to use the filter just to hook in an action.